js/deletePost.js

```diff
@@ -0,0 +1,44 @@
+import { API_BASE_URL } from "./const.mjs";
+
+const modal = document.getElementById("myModal");
+const deleteButton = document.getElementById("deletePostButton");
+
+async function deletePost(postId) {
+  const token = localStorage.getItem("accessToken");
+  const url = `${API_BASE_URL}social/posts/${postId}`;
+  // console.log(url);
+
+  try {
+    const response = await fetch(url, {
+      method: "DELETE",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${token}`,
+      },
+      body: JSON.stringify({
```

Comment:
According to the API doc., this is incorrect. The id is meant to be passed to the endpoint in the url lik `/social/posts/<id>`